cpp/net/connection_pool.cc

```diff
@@ -90,12 +229,14 @@ unique_ptr<ConnectionPool::Connection> ConnectionPool::Get(const URL& url) {
 
   if (it == conns_.end() || it->second.empty()) {
     VLOG(1) << "new evhtp_connection for " << key.first << ":" << key.second;
-    unique_ptr<ConnectionPool::Connection> conn(
-        new Connection(base_->HttpConnectionNew(key.first, key.second),
-                       move(key)));
-    evhtp_set_hook(&conn->connection()->hooks, evhtp_hook_on_connection_fini,
+    unique_ptr<ConnectionPool::Connection> conn(new Connection(
+        url.Protocol() == "https"
+            ? base_->HttpsConnectionNew(key.first, key.second, ssl_ctx_.get())
```

Comment:
Agreed, that's why the big Easy Spot/Remove `----8<----` section is in there (or, rather, in the actual blocking method), along with a reference to the libevt PR which will Do It Right - I'll ping that guy and see if he's had any luck on that front.